src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java

```diff
@@ -55,6 +60,32 @@ public UserCommons getUserCommonsById(
     return userCommons;
   }
 
+  @ApiOperation(value = "Get all user commons with a specific commonsId")
+  @GetMapping("/allwithcommonsid")
+  public Iterable<UserCommons> getAllUserCommonsByCommonsId(
+    @ApiParam("commonsId") @RequestParam Long commonsId) throws JsonProcessingException {
+      boolean isAdmin = false;
+
+      for(GrantedAuthority a : getCurrentUser().getRoles()){
+        // comparing if they are equal does NOT work.
+        // could be padding issues?
+        if(a.getAuthority().contains("ROLE_ADMIN")){
+          isAdmin = true;
+        }
+      }
```

Comment:
It might be good to factor this out into a separate method, both for readability and reusability.

Reply:
Maybe a method in `https://github.com/ucsb-cs156-s22/s22-5pm-happycows/blob/main/src/main/java/edu/ucsb/cs156/happiercows/services/CurrentUserServiceImpl.java`
 
```java
 public static boolean isAdmin() {
       for(GrantedAuthority a : getCurrentUser().getRoles()){
              if(a.getAuthority().contains("ROLE_ADMIN")){
                  return true;
        }
       return false; 
      }
 }
```

or even better:

```java
 public static boolean hasRole(String role) {
       for(GrantedAuthority a : getCurrentUser().getRoles()){
              if(a.getAuthority().contains(role){
                  return true;
        }
       return false; 
      }
 }
 
  public static boolean isAdmin() {
     return hasRole("ROLE_ADMIN");
  }
```